eris.go

```diff
@@ -66,6 +67,10 @@ func wrap(err error, msg string) error {
 			}
 		}
 	case *wrapError:
+		// insert the frame into the stack
+		if root, ok := Cause(err).(*rootError); ok {
```

Comment:
i'm thinking we could optimize some parts of the code by storing a `root error` or `cause error` field in wrap errors that directly points to the root error. it would save us some looping / calling to `Cause` and `Unwrap` and it's not much additional data to pass around. doesn't need to happen right now though unless you think it's worthwhile.

Reply:
ya, while writing this function I wish I had a way to directly access the rootErr so I am open to that idea.

Reply:
planning to merge this without it but we can look into adding that in the near future